packages/frontend/src/utils/fetch.ts

```diff
@@ -2,8 +2,11 @@ import type { AuthRoutes, InferAuthRouteResult, InferAuthRouteBody } from '@chat
 import type { Payload } from '@hapi/boom';
 
 export class APIError extends Error {
-	public constructor(public readonly payload: Payload) {
+	public readonly payload: Payload | undefined; // can be undefined in case of network error
```

Comment:
> // can be undefined in case of network error

No, it never will be. We only throw it ourselves for non-2xx, in case of a network error you'll get a different error class. See my review on `useLoggedInUser`